util/data/msgencode.c

```diff
@@ -479,11 +483,10 @@ packed_rrset_encode(struct ub_packed_rrset_key* key, sldns_buffer* pkt,
 				return r;
 			sldns_buffer_write(pkt, &key->rk.type, 2);
 			sldns_buffer_write(pkt, &key->rk.rrset_class, 2);
-			if(data->rr_ttl[j] < timenow)
+			if(data->rr_ttl[j] < adjust)
```

Comment:
Is this intentionally inconsistent with the checks in packed_rrset.c (where the comparison is still `d->rr_ttl[i] < now`)?

Reply:
The reasoning was that here the original code performed an adjustment depending on the flag `PACKED_RRSET_FIXEDTTL` being set. This adjustment changed an input parameter to the function (`timenow`). I made this explicit, because the change for the new feature would add additional decision logic here, and it felt wrong to use an input variable rather than something explicit. If you would prefer this to be explicit everywhere, I can also change the code in `packed_rrset.c`, let me know.

Reply:
Yes I would like you to also do it like that in packed_rrset.c, because I want to make sure that it does not subtract lower than 0, so the compare < adjust is better than < now to make really sure.  Because otherwise it could create negative, high-bit-set TTLs, and those are considered a resolver problem.

Reply:
I addressed this in commit d253db0, can you check if that meets your expectations?

Reply:
Okay from me!